File: server/importer.test.ts
import assert from "node:assert/strict";
import { describe, it } from "node:test";
import { parseAccountImport } from "./importer";

describe("parseAccountImport", () => {
  it("解析 Tab 分隔账号", () => {
    const result = parseAccountImport("user@example.com\tpass\tclient-id\trefresh-token");
    assert.equal(result.errors.length, 0);
    assert.equal(result.accounts[0].email, "user@example.com");
  });

  it("解析四横线分隔账号并保留令牌中的分隔符", () => {
    const result = parseAccountImport("user@example.com----pass----client-id----token----tail");
    assert.equal(result.errors.length, 0);
    assert.equal(result.accounts[0].refreshToken, "token----tail");
  });

  it("报告无效行号", () => {
    const result = parseAccountImport("bad-line\nuser@example.com----p----c----t");
    assert.equal(result.errors[0].line, 1);
    assert.equal(result.accounts.length, 1);
  });
});

File: server/internal/importer/importer_test.go
package importer

import (
	"testing"

	"github.com/amine123max/Mail/server/internal/model"
)

func TestParseTabSeparated(t *testing.T) {
	result := Parse("user@example.com\tpass\tclient-id\trefresh-token")
	if len(result.Errors) != 0 || len(result.Accounts) != 1 || result.Accounts[0].Email != "user@example.com" {
		t.Fatalf("unexpected parse result: %#v", result)
	}
}

func TestParseHyphenTokenTail(t *testing.T) {
	result := Parse("user@example.com----pass----client-id----token----tail")
	if len(result.Errors) != 0 || result.Accounts[0].RefreshToken != "token----tail" {
		t.Fatalf("unexpected parse result: %#v", result)
	}
}

func TestSerialize(t *testing.T) {
	content := Serialize([]model.AccountCredentials{{Email: "first@example.com", Password: "pass", ClientID: "client", RefreshToken: "refresh----tail"}})
	if content != "first@example.com----pass----client----refresh----tail" {
		t.Fatalf("unexpected export: %q", content)
	}
}
